sdks/python/apache_beam/io/gcp/gcsio.py

```diff
@@ -172,7 +173,7 @@ def get_bucket(self, bucket_name, **kwargs):
     try:
       return self.client.lookup_bucket(
           bucket_name, retry=self._storage_client_retry, **kwargs)
-    except NotFound:
+    except (Forbidden, NotFound):
```

Comment:
Yeah, you're right about it being exists, I added it there as well. 

> The Forbidden error mentions it can be a permission issue or the bucket does not exist. So we might swallow actual permission issues too.

I think this is going to be true even if we do what you're suggesting. If there are permissions issues, the bucket won't show up in our list call either